### ISSUE TITLE
Trouble with Imagemagick's pipe on windows

### ISSUE THREAD

**Issue by Ser-Gen** [mozilla/mozjpeg#269]
Can't use in-memory compression with pipe.

Command from this article:
https://calendar.perfplanet.com/2014/mozjpeg-3-0/

```
convert input.png TGA:- | cjpeg -outfile output.jpg -targa
```

With this image
![input](https://user-images.githubusercontent.com/1856145/31141256-8cd97630-a87f-11e7-90c5-9e2f7614a2d7.png)
I got broken

![output](https://user-images.githubusercontent.com/1856145/31141311-b686918e-a87f-11e7-8812-a6bca9c34e82.jpg)

I have tried other modifiers `JPEG:-`, `PNG:-`, output:

```
Invalid or unsupported Targa file
```

```
Windows 10

> cjpeg -v
mozjpeg version 3.2 (build 20170722)

> convert -version
Version: ImageMagick 6.9.9-5 Q16 x64 2017-08-03 http://www.imagemagick.org
```

**Comment by pornel** [comment]
Try 

```sh
convert input.png BMP3:- | cjpeg -outfile output.jpg 
``

**Comment by Ser-Gen** [comment]
Nope, `BMP2` and `BMP3` outputs bad image, same as `TGA`.

```
convert input.png BMP3:- | cjpeg -outfile output.jpg
```

![output](https://user-images.githubusercontent.com/1856145/31182632-169be332-a92d-11e7-88e5-16f58c9b2f49.jpg)

**Comment by pornel** [comment]
In that case, are you sure imagemagick works correctly? It's quite unusual for 3 different formats to fail in an identical way.

**Comment by Ser-Gen** [comment]
I will try to check IM work somehow.

By the way, some fails a bit different.

```
> convert input.png BMP2:- | cjpeg -outfile output.bmp2.jpg
> convert input.png BMP3:- | cjpeg -outfile output.bmp3.jpg
> convert input.png JPEG:- | cjpeg -outfile output.jpeg.jpg
Unrecognized input file format --- perhaps you need -targa
> convert input.png JPEG:- | cjpeg -outfile output.jpeg.jpg -targa
Invalid or unsupported Targa file
> convert input.png PNG:- | cjpeg -outfile output.png.jpg
Unrecognized input file format --- perhaps you need -targa
> convert input.png PNG:- | cjpeg -outfile output.png.jpg -targa
Invalid or unsupported Targa file
> convert input.png TGA:- | cjpeg -outfile output.tga.jpg

473 output.bmp2.jpg
473 output.bmp3.jpg
0 output.jpeg.jpg
0 output.png.jpg
485 output.tga.jpg
```

**Comment by Ser-Gen** [comment]
Yep, IM pipe is broken :\\
Thank you!

**Comment by JoshuaCode** [comment]
@Ser-Gen did you ever resolve your broken pipe issue with ImageMagick? I am experiencing the same issue trying to pipe from ImageMagick to cjpeg.

**Comment by JoshuaCode** [comment]
I solved my issue. I was using a PowerShell CLI and it turns out that PowerShell corrupts binary data when passed thru STDIN or STDOUT. I switched to command prompt (cmd.exe) and the output was no longer corrupted. PowerShell's issue with processing binary data is a known issue, and there are some work arounds, but the easiest is not using PowerShell. See links below for details.

I did run into another issue piping the output from ImageMagick to cjpeg that I wanted to document for others that may come across this. The following command was causing the error message "Premature end of input file".

`magick convert -resize 75 input.jpg pnm:- | cjpeg -outfile output.jpg`

By default, the input stream is buffered. To ensure information on the source file or terminal is read as soon as its available, set the buffer size to 0 by adding '-define stream:buffer-size=0'. The final working command is 

`magick convert -define stream:buffer-size=0 -resize 75 input.jpg pnm:- | cjpeg -outfile output.jpg`

- MS Docs - [un-messing Unicode in PowerShell](https://docs.microsoft.com/en-us/archive/blogs/sergey_babkins_blog/un-messing-unicode-in-powershell)
- PowerShell GitHub Repo - [Don’t parse the pipeline as text when it is directed from an EXE to another EXE or file. Keep the bytes as-is. #1908](https://github.com/PowerShell/PowerShell/issues/1908)
- Windows PowerShell Cookbook - [Capture and Redirect Binary Process Output](https://www.powershellcookbook.com/recipe/WCiL/capture-and-redirect-binary-process-output)
- [PowerShell’s Object Pipeline Corrupts Piped Binary Data](https://brianreiter.org/2010/01/29/powershells-object-pipeline-corrupts-piped-binary-data/)
